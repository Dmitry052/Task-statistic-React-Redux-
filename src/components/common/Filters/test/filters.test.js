import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-localstorage-mock';

import { Filters } from './../index';

configure({ adapter: new Adapter() });

describe('<Filters />', () => {
  const queryParams = {
    date: {
      testDate: '2017-01-01'
    },
    testString: 'test string'
  };
  const filters = [
    {
      name: 'testString',
      type: 'string',
      displayName: 'TEST-STRING'
    },
    {
      name: 'testDate',
      type: 'date',
      displayName: 'TEST-DATE'
    },
    {
      name: 'testSelectx',
      type: 'selectx',
      displayName: 'TEST-SELECTX'
    }
  ];

  it('render component with role "super', () => {
    const role = 'super';
    const tree = shallow(
      <Filters
        role={role}
        filters={filters}
        queryParams={queryParams}
        onChangeDate={() => {}}
        handleUpdateData={() => {}}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('render component with role "user', () => {
    const role = 'super';
    const tree = shallow(
      <Filters
        role={role}
        filters={filters}
        queryParams={queryParams}
        onChangeDate={() => {}}
        handleUpdateData={() => {}}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('render component without role', () => {
    localStorage.setItem('role', 'user');
    const tree = shallow(
      <Filters
        filters={filters}
        queryParams={queryParams}
        onChangeDate={() => {}}
        handleUpdateData={() => {}}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
