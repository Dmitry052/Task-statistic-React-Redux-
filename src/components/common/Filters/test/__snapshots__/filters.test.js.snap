// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> render component with role "super 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testString-string"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-STRING
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      field="testString"
      id="TEST-STRING-0"
      onChange={[Function]}
      type="string"
      value="test string"
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testDate-date"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-DATE
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      field="testDate"
      id="TEST-DATE-1"
      onChange={[Function]}
      type="date"
      value="2017-01-01"
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testSelectx-selectx"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-SELECTX
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="select"
      field="testSelectx"
      id="TEST-SELECTX-2"
      onChange={[Function]}
    >
      <option
        value={null}
      />
    </FormControl>
  </Col>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    onClick={[Function]}
  >
    Update
  </Button>
</Row>
`;

exports[`<Filters /> render component with role "user 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testString-string"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-STRING
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      field="testString"
      id="TEST-STRING-0"
      onChange={[Function]}
      type="string"
      value="test string"
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testDate-date"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-DATE
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      field="testDate"
      id="TEST-DATE-1"
      onChange={[Function]}
      type="date"
      value="2017-01-01"
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testSelectx-selectx"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-SELECTX
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="select"
      field="testSelectx"
      id="TEST-SELECTX-2"
      onChange={[Function]}
    >
      <option
        value={null}
      />
    </FormControl>
  </Col>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    onClick={[Function]}
  >
    Update
  </Button>
</Row>
`;

exports[`<Filters /> render component without role 1`] = `
<Row
  bsClass="row"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testString-string"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-STRING
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      field="testString"
      id="TEST-STRING-0"
      onChange={[Function]}
      type="string"
      value="test string"
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testDate-date"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-DATE
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="input"
      field="testDate"
      id="TEST-DATE-1"
      onChange={[Function]}
      type="date"
      value="2017-01-01"
    />
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    key="filter-testSelectx-selectx"
    lg={2}
    md={3}
    xs={4}
  >
    <ControlLabel
      bsClass="control-label"
      srOnly={false}
    >
      TEST-SELECTX
    </ControlLabel>
    <FormControl
      bsClass="form-control"
      componentClass="select"
      field="testSelectx"
      id="TEST-SELECTX-2"
      onChange={[Function]}
    >
      <option
        value={null}
      />
    </FormControl>
  </Col>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    onClick={[Function]}
  >
    Update
  </Button>
</Row>
`;
